src/settings.c

```diff
@@ -0,0 +1,66 @@
+/* settings.c -- global settings.
+ *
+ * Copyright (c) 2020, James Duong <duong.james@gmail.com> All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of LCUI nor the names of its contributors may be used
+ *     to endorse or promote products derived from this software without
+ *     specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <LCUI_Build.h>
+#include <LCUI/types.h>
+#include <LCUI/util.h>
+#include <LCUI/main.h>
+#include <LCUI/settings.h>
+
+static LCUI_SettingsRec self;
+static LCUI_SysEventRec event;
```

Comment:
`event` should not be a global variable because it is rarely used, it is recommended to use it as a local variable.

Reply:
I'm not clear on what the lifetime of an event passed to LCUI_TriggerEvent needs to be. It looks like we pass it by pointer, then LCU_TriggerEvent copies the pointer only, then notifies listeners and no longer needs it.

If that's the case, I'll just make it a local variable.

Reply:
@jduo `LCUI_TriggerEvent()` will directly run the event handler, which is called synchronously. You don't need to worry about the life cycle of `event` variable.